### PR TITLE
feat: Tuning Resource Validation Check

### PR DESCRIPTION
**Pull request by ishaansehgal99**
**Reason for Change**:
Tuning Resource Validation Check

### CODE REVIEW

File: api/v1alpha1/workspace_validation.go [hunk 1]
```diff
@@ -299,16 +300,23 @@ func (r *DataDestination) validateUpdate(old *DataDestination) (errs *apis.Field
 	return errs
 }
 
-func (r *ResourceSpec) validateCreate(inference InferenceSpec) (errs *apis.FieldError) {
+func (r *ResourceSpec) validateCreateWithTuning(tuning *TuningSpec) (errs *apis.FieldError) {
+	if tuning != nil && *r.Count > 1 {
```
Comment (Fei-Guo) [inline api/v1alpha1/workspace_validation.go:304]: Nit: tuning != nil is not needed.

File: api/v1alpha1/workspace_validation.go [hunk 2]
```diff
@@ -299,16 +300,23 @@ func (r *DataDestination) validateUpdate(old *DataDestination) (errs *apis.Field
...
+		errs = errs.Also(apis.ErrInvalidValue("Tuning does not currently support multinode configurations. Please set the node count to 1. Future support with DeepSpeed will allow this.", "count"))
+	}
+	return errs
+}
+
+func (r *ResourceSpec) validateCreateWithInference(inference *InferenceSpec) (errs *apis.FieldError) {
 	var presetName string
-	if inference.Preset != nil {
+	if inference != nil && inference.Preset != nil {
```
Comment (Fei-Guo) [inline api/v1alpha1/workspace_validation.go:312]: Caller side already does the check, inference != nil  is not needed.

Comment (ishaansehgal99) [reply api/v1alpha1/workspace_validation.go:312]: Fixed

File: api/v1alpha1/workspace_validation.go (hunk 1 above)
Comment (ishaansehgal99) [reply api/v1alpha1/workspace_validation.go:304]: resolved

**Review by Fei-Guo — APPROVED** [review]

**Pull request merged**